Source/HBIOS/ps2iface.inc

```diff
@@ -0,0 +1,247 @@
+;__________________________________________________________________________________________________
+; HARDWARE INTERFACE FOR POLLED AND INTERRUPT DRIVEN PS2 KEYBOARD.
+;
+; IN POLLED MODE THE KEYBOARD DEVICE IS IDENTIFIED BY THE TABLE POINTED TO BY IY
+; IN INTERRUPT MODE THE KEYBOARD DEVICE IS UNKNOWN SO PREDEFINED PORT VALUES ARE USED.

```

Comment:
It would be preferable to use IY in this file and set IY in the interrupt routine itself.  Take a look at sio.asm starting at line 245.  If only one KBD device can exist with interrupts enabled, then IY should still be set at the start of the interrupt handler and IY should be used in this file to get the port numbers.

Reply:
Thanks Wayne.  I didn't know there was a way to resolve this. Will take a look.